clang/test/Sema/bitint-bitfield-promote.c

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -fsyntax-only -verify -std=c23 %s
+
+// GH87641 noticed that integer promotion of a bit-field of bit-precise integer
+// type was promoting to int rather than the type of the bit-field.
+struct S {
+  unsigned _BitInt(7) x : 2;
+  unsigned _BitInt(2) y : 2;
+  unsigned _BitInt(72) z : 28;
```

Comment:
Is there a reason all of these are unsigned? Maybe adding some tests for signed `_BitInt`s as well might be a good idea—or do we have those already?

Reply:
Wouldn't mind a 31 and a 33 as well.

Reply:
Sure, I can add some signed tests with different boundaries!